internal/resources/slo/resource_slo.go

```diff
@@ -221,6 +221,22 @@ Resource manages Grafana SLOs.
 								},
 							},
 						},
+						"advanced_options": {
+							Type:        schema.TypeList,
+							MaxItems:    1,
```

Comment:
question: are there pros/cons to omitting this `MaxItems` field?

I'm thinking about what happens if we add another field under advancedOptions, months from now. Do we end up with the same chicken and egg issue where older terraform provider yells about new fields that it doesn't know about.

Reply:
I assumed this was the number of advanced_options, not the number of sub items when I did this, but your right it shouldn't have a max items restriction.

Reply:
For example the top level alerting has MaxItems set to 1.
`			"alerting": {
				Type:     schema.TypeList,
				MaxItems: 1,
				Optional: true,
				Description: `Configures the alerting rules that will be generated for each
				time window associated with the SLO. Grafana SLOs can generate
				alerts when the short-term error budget burn is very high, the
				long-term error budget burn rate is high, or when the remaining
				error budget is below a certain threshold. Annotations and Labels support templating.`,
				Elem: &schema.Resource{
					Schema: map[string]*schema.Schema{
						"label": {
							Type:        schema.TypeList,
							Optional:    true,
							Description: `Labels will be attached to all alerts generated by any of these rules.`,
							Elem:        keyvalueSchema,
						},
						"annotation": {
							Type:        schema.TypeList,
							Optional:    true,
							Description: `Annotations will be attached to all alerts generated by any of these rules.`,
							Elem:        keyvalueSchema,
						},
						"fastburn": {
							Type:        schema.TypeList,
							Optional:    true,
							MaxItems:    1,
							Description: "Alerting Rules generated for Fast Burn alerts",
							Elem: &schema.Resource{
								Schema: map[string]*schema.Schema{
									"label": {
										Type:        schema.TypeList,
										Optional:    true,
										Description: "Labels to attach only to Fast Burn alerts.",
										Elem:        keyvalueSchema,
									},
									"annotation": {
										Type:        schema.TypeList,
										Optional:    true,
										Description: "Annotations to attach only to Fast Burn alerts.",
										Elem:        keyvalueSchema,
									},
								},
							},
						},
						"slowburn": {
							Type:        schema.TypeList,
							MaxItems:    1,
							Optional:    true,
							Description: "Alerting Rules generated for Slow Burn alerts",
							Elem: &schema.Resource{
								Schema: map[string]*schema.Schema{
									"label": {
										Type:        schema.TypeList,
										Optional:    true,
										Description: "Labels to attach only to Slow Burn alerts.",
										Elem:        keyvalueSchema,
									},
									"annotation": {
										Type:        schema.TypeList,
										Optional:    true,
										Description: "Annotations to attach only to Slow Burn alerts.",
										Elem:        keyvalueSchema,
									},
								},
							},
						},
						"advanced_options": {
							Type:        schema.TypeList,
							MaxItems:    1,
							Optional:    true,
							Description: "Advanced Options for Alert Rules",
							Elem: &schema.Resource{
								Schema: map[string]*schema.Schema{
									"min_failures": {
										Type:         schema.TypeInt,
										Optional:     true,
										Description:  "Minimum number of failed events to trigger an alert",
										ValidateFunc: validation.IntAtLeast(0),
									},
								},
							},
						},
					},
				},
			}`